flixel/input/gamepad/ButtonIndex.hx

```diff
@@ -0,0 +1,405 @@
+			case LEFT_STICK_BTN: PS4ButtonID.LEFT_STICK_BTN;
+			case RIGHT_STICK_BTN: PS4ButtonID.RIGHT_STICK_BTN;
+			case LEFT_SHOULDER: PS4ButtonID.L1;
+			case RIGHT_SHOULDER: PS4ButtonID.R1;
+			case LEFT_TRIGGER: PS4ButtonID.L2;
+			case RIGHT_TRIGGER: PS4ButtonID.R2;
+			//case DPAD_UP: 
+			//case DPAD_DOWN: 
+			//case DPAD_LEFT: 
+			//case DPAD_RIGHT: 
+			default: -1;
+		}
+	}
+	
+	public inline function getRawPS3(buttonID:ButtonID):Int
+	{
+		return switch(buttonID)
+		{
+			case A: PS3ButtonID.X;
+			case B: PS3ButtonID.CIRCLE;
+			case X: PS3ButtonID.SQUARE;
+			case Y: PS3ButtonID.TRIANGLE;
+			case BACK: PS3ButtonID.SELECT;
+			case GUIDE: PS3ButtonID.PS;
+			case START: PS3ButtonID.START;
+			case LEFT_STICK_BTN: PS3ButtonID.LEFT_STICK_BTN;
+			case RIGHT_STICK_BTN: PS3ButtonID.RIGHT_STICK_BTN;
+			case LEFT_SHOULDER: PS3ButtonID.L1;
+			case RIGHT_SHOULDER: PS3ButtonID.R1;
+			case LEFT_TRIGGER: PS3ButtonID.L2;
+			case RIGHT_TRIGGER: PS3ButtonID.R2;
+			case DPAD_UP: PS3ButtonID.DPAD_UP;
+			case DPAD_DOWN: PS3ButtonID.DPAD_DOWN;
+			case DPAD_LEFT: PS3ButtonID.DPAD_LEFT;
+			case DPAD_RIGHT: PS3ButtonID.DPAD_RIGHT;
+			default: -1;
+		}
+	}
+	
+	public inline function getRawXbox(buttonID:ButtonID):Int
+	{
+		return switch(buttonID)
+		{
+			case A: XboxButtonID.A;
+			case B: XboxButtonID.B;
+			case X: XboxButtonID.X;
+			case Y: XboxButtonID.Y;
+			case BACK: XboxButtonID.BACK;
+			case GUIDE: XboxButtonID.XBOX;
+			case START: XboxButtonID.START;
+			case LEFT_STICK_BTN: XboxButtonID.LEFT_STICK_BTN;
+			case RIGHT_STICK_BTN: XboxButtonID.RIGHT_STICK_BTN;
+			case LEFT_SHOULDER: XboxButtonID.LB;
+			case RIGHT_SHOULDER: XboxButtonID.RB;
+			case DPAD_UP: XboxButtonID.DPAD_UP;
+			case DPAD_DOWN: XboxButtonID.DPAD_DOWN;
+			case DPAD_LEFT: XboxButtonID.DPAD_LEFT;
+			case DPAD_RIGHT: XboxButtonID.DPAD_RIGHT;
+			
+			case LEFT_TRIGGER:  XboxButtonID.LEFT_TRIGGER;
+			case RIGHT_TRIGGER: XboxButtonID.RIGHT_TRIGGER;
+			
+			default: -1;
+		}
+	}
+	
+	public inline function getRawXInput(buttonID:ButtonID):Int
+	{
+		return switch(buttonID)
+		{
+			case A: XInputButtonID.A;
+			case B: XInputButtonID.B;
+			case X: XInputButtonID.X;
+			case Y: XInputButtonID.Y;
+			case BACK: XInputButtonID.BACK;
+			case GUIDE: XInputButtonID.GUIDE;
+			case START: XInputButtonID.START;
+			case LEFT_STICK_BTN: XInputButtonID.LEFT_STICK_BTN;
+			case RIGHT_STICK_BTN: XInputButtonID.RIGHT_STICK_BTN;
+			case LEFT_SHOULDER: XInputButtonID.LB;
+			case RIGHT_SHOULDER: XInputButtonID.RB;
+			case DPAD_UP: XInputButtonID.DPAD_UP;
+			case DPAD_DOWN: XInputButtonID.DPAD_DOWN;
+			case DPAD_LEFT: XInputButtonID.DPAD_LEFT;
+			case DPAD_RIGHT: XInputButtonID.DPAD_RIGHT;
+			
+			case LEFT_TRIGGER: XInputButtonID.LEFT_TRIGGER;
+			case RIGHT_TRIGGER: XInputButtonID.RIGHT_TRIGGER;
+			
+			//case LEFT_TRIGGER: 
+			//case RIGHT_TRIGGER: 
+			default: -1;
+		}
+	}
+	
+	/**************/
+	
+	public inline function getBtnOUYA(rawID:Int):ButtonID
```

Comment:
I think it's better to lose the inline here. Inline makes sense for very short functions without complex control structures (loops, switch-case, even if-else I think) that are called often. None of that is the case here I think?